### PR TITLE
Themes: Add current-theme query component

### PR DESCRIPTION
**Pull request by seear**
A [query component](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#query-components) to replace the old-style fetcher component. Will make it very easy to determine current theme properly in theme sheets (#5285)

No visual changes

screen shot 2016-05-13 at 11 56 20

**To Test**
* Go to http://calypso.localhost:3000/design/
* choose a site
Expected: the _CURRENT THEME_ bar works as previously
Things to check:
* switching sites
* jetpack site
* `localStorage.clear();` in console then refresh
* activate a theme

### CODE REVIEW

**Comment by ockham** [comment]
The (visual) current theme component isn't updated for me when activating a theme on a Jetpack site, and even the thanks modal display's the previous theme's name :-(

**Comment by ockham** [comment]
Actually, not just on Jetpack sites...

File: client/state/themes/current-theme/reducer.js [hunk 1]
```diff
@@ -13,30 +13,39 @@ import {
...
+			//TODO: show notification
+			return state.setIn( [ 'requesting', action.siteId ], false );
 		case THEME_ACTIVATE:
 			return state.set( 'isActivating', true );
 		case THEME_ACTIVATED:
 			return state
 				.set( 'isActivating', false )
 				.set( 'hasActivated', true )
-				.setIn( [ 'currentThemes', action.site.ID ], action.theme );
+				.setIn( [ 'currentThemes', action.siteId ], action.theme );
```
Comment (seear) [inline client/state/themes/current-theme/reducer.js]: @ockham: this is the problem—overzealous triming of `site` usage. Fixing...

**Comment by seear** [comment]
> The (visual) current theme component isn't updated for me when activating a theme on a Jetpack site, and even the thanks modal display's the previous theme's name :-(

fixed in b2ba232

**Comment by ockham** [comment]
Loving it. Squash it, :ship: it!

Aside — Can we incorporate `ActivatingThemeData`’s functionality into this component as a follow-up, and drop that one? :-)

**Comment by seear** [comment]
> Aside — Can we incorporate ActivatingThemeData’s functionality into this component as a follow-up, and drop that one? :-)

Yeah, was mulling that over myself, definite next step.

Thanks for review :)

**Pull request merged**